Automattic/wp-calypso
# Packages: Fix components and search storybook

#### Changes proposed in this Pull Request

This PR fixes storybook instances of the components and search packages. Not sure how it worked before, but running storybook now errors out with some missing modules (`ts-loader` and `react-docgen-typescript-loader`). So this PR adds those in custom storybook webpack configs. In addition, we remove a `emitDeclarationOnly` and simplify the way we do the mobile viewport check, to avoid extra complications from importing `@automattic/viewport`, `@wordpress/viewport` or `useViewportMatch` from `@wordpress/compose`.

Aims to unblock #46825.

#### Testing instructions

* Run `yarn`
* Run `yarn run components:storybook:start` and verify it builds and works
* Run `yarn run search:storybook:start`  and verify it builds and works

#### Next steps

I think we can improve the webpack configs a bit to improve build performance, and introduce some caching, but this should do the job for now.
